services/server/test/integration/repository-handlers/files.spec.ts

```diff
@@ -10,370 +10,409 @@ import { LocalChainFixture } from "../../helpers/LocalChainFixture";
-        `http://localhost:10000contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/metadata.json`,
-      ]);
-    const res4 = await agent.get(`/files/contracts/${chainFixture.chainId}`);
-    chai.expect(res4.body).to.deep.equal({
-      full: [chainFixture.defaultContractAddress],
-      partial: [],
-    });
+  const serverFixtureWithDatabase = new ServerFixture();
+  serverFixtureWithDatabase.identifier = StorageIdentifiers.SourcifyDatabase;
+  const serverFixtureWithRepositoryV1 = new ServerFixture({
+    port: 5556, // use a different port
+    read: StorageIdentifiers.RepositoryV1,
+    writeOrErr: [StorageIdentifiers.RepositoryV1],
   });
+  serverFixtureWithRepositoryV1.identifier = StorageIdentifiers.RepositoryV1;
+
+  [serverFixtureWithDatabase, serverFixtureWithRepositoryV1].forEach(
+    (serverFixture) => {
+      it(`should fetch files of specific address, using a non-checksummed address. Storage type: ${serverFixture.identifier}`, async function () {
+        const agent = chai.request.agent(serverFixture.server.app);
+        // Wait for the server to complete the previous contract verification
+        await waitSecs(1);
+        await agent
+          .post("/")
+          .field("address", chainFixture.defaultContractAddress)
+          .field("chain", chainFixture.chainId)
+          .attach(
+            "files",
+            chainFixture.defaultContractMetadata,
+            "metadata.json"
+          )
+          .attach("files", chainFixture.defaultContractSource, "Storage.sol");
+        const res0 = await agent.get(
+          `/files/${
+            chainFixture.chainId
+          }/${chainFixture.defaultContractAddress.toLocaleLowerCase()}`
+        );
+        chai.expect(res0.body).to.include.deep.members([
+          {
+            name: "Storage.sol",
+            path: `contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/sources/project:/contracts/Storage.sol`,
+            content: chainFixture.defaultContractSource.toString(),
+          },
+          {
+            name: "metadata.json",
+            path: `contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/metadata.json`,
+            content: chainFixture.defaultContractMetadata.toString(),
+          },
+        ]);
+        const res1 = await agent.get(
+          `/files/tree/any/${
+            chainFixture.chainId
+          }/${chainFixture.defaultContractAddress.toLocaleLowerCase()}`
+        );
+        chai.expect(res1.body.status).to.equal("full");
+        chai
+          .expect(res1.body.files)
+          .to.have.members([
+            `http://localhost:10000/contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/sources/project:/contracts/Storage.sol`,
+            `http://localhost:10000/contracts/full_match/${chainFixture.chainId}/${chainFixture.defaultContractAddress}/metadata.json`,
+          ]);
```

Comment:
Both old and new are valid but I wondered why we are doing the changes here

Reply:
What changes?

Reply:
See the old code https://github.com/ethereum/sourcify/blob/e1ef4117fb6919b8e21eb2e357e2e2a75ae3d35d/services/server/test/integration/repository-handlers/files.spec.ts#L52-L58

Reply:
Ah ok, in the old code I wrongly implemented this test `http://localhost:10000contracts`, there is a missing `/`